currencies/rate_converter.go

```diff
@@ -160,14 +168,25 @@ func (rc *RateConverter) LastUpdated() time.Time {
 
 // Rates returns current conversions rates
 func (rc *RateConverter) Rates() Conversions {
-	if rc.constantRates != nil {
-		// Converter is not active, returning the constant rates
-		return rc.constantRates
-	}
-	if rates := rc.rates.Load(); rates != nil {
+	if rates := rc.rates.Load(); rates != (*Rates)(nil) && rates != nil {
 		return rates.(*Rates)
 	}
-	return nil
+	return rc.constantRates
+}
+
+// CheckStaleRates checks if loaded third party conversion rates are stale
+func (rc *RateConverter) CheckStaleRates() bool {
+	if rc.staleRatesThreshold <= 0 {
+		return false
+	}
+	currentTime := time.Now().UTC()
```

Comment:
If you factor out the `time.Now()`call to a dependency, you can mock the time and avoid the sleeps in the tests.

Reply:
👍 will handle this in the currency rate refactor PR.